### PR TITLE
Feature/fortran ops lazy

### PR DESCRIPTION
**Pull request by Ashutosh-Londhe**
Enabled code generation for OPS_LAZY implementation at Fortran side for sequential and MPI version

### CODE REVIEW

File: makefiles/Makefile.f_app [hunk 1]
```diff
@@ -26,7 +26,7 @@ all: $(TARGETS)
 cleanfiles:
 	rm -f *.o *.mod
 
-$(APP)_seq: .generated $(OPS_FILES_GEN) Makefile
+$(APP)_seq: .generated $(OPS_FILES_GEN) Makefile cleanfiles
```
Comment (reguly) [inline makefiles/Makefile.f_app:29]: let's remove this cleanfiles dependency

File: ops/c/src/core/ops_lazy.cpp [hunk 2]
```diff
@@ -171,8 +171,12 @@ size_t ops_internal_get_cache_size(OPS_instance *instance) {
 /////////////////////////////////////////////////////////////////////////
 
 void ops_enqueue_kernel(ops_kernel_descriptor *desc) {
+
+  // printf("Inside enqueue: \n");  
+  // printf("name: %s, dim: %d, index: %d, isdevise: %d block_dim: %d\n", desc->name, desc->dim, desc->index, desc->isdevice, desc->block->dims);
```
Comment (reguly) [inline ops/c/src/core/ops_lazy.cpp:176]: Remove these, or in a conditional checking OPS_diags>4

File: ops/c/src/core/ops_lazy.cpp [hunk 3]
```diff
@@ -931,18 +939,63 @@ void ops_execute(OPS_instance *instance) {
...
     ops_kernel_list[i] = nullptr;
   }
   ops_kernel_list.clear();
 }
 
+void create_kerneldesc_and_enque(char const *name, ops_arg *args, int nargs, int index, int dim, int isdevice, int *range, ops_block block, void (*func)(struct ops_kernel_descriptor *desc))
+{
+    ops_kernel_descriptor *desc = (ops_kernel_descriptor *)calloc(1,sizeof(ops_kernel_descriptor));
+
+    desc->name = (char*) calloc(strlen(name)+1, sizeof(char));
```
Comment (reguly) [inline ops/c/src/core/ops_lazy.cpp:956]: free this

File: ops/c/src/core/ops_lazy.cpp [hunk 4]
```diff
@@ -931,18 +939,63 @@ void ops_execute(OPS_instance *instance) {
...
+    desc->args = (ops_arg*)ops_malloc(nargs*sizeof(ops_arg));
+
+    for ( int n=0; n < nargs; n++) {
+        desc->args[n] = args[n];
+
+        if (args[n].argtype == OPS_ARG_DAT)
+            desc->hash = ((desc->hash << 5) + desc->hash) + args[n].dat->index;
+        if (args[n].argtype == OPS_ARG_GBL && args[n].acc == OPS_READ) {
+            char *tmp = (char*)ops_malloc(args[n].dim*args[n].elem_size);
+            memcpy(tmp, args[n].data,args[n].dim*args[n].elem_size);
```
Comment (reguly) [inline ops/c/src/core/ops_lazy.cpp:985]: is this correctly freed?

**Review by reguly — CHANGES_REQUESTED** [review]

File: ops/c/src/core/ops_lazy.cpp (hunk 4 above)
Comment (Ashutosh-Londhe) [reply ops/c/src/core/ops_lazy.cpp:985]: https://github.com/OP-DSL/OPS/blob/e586f89b410eb2ed164ad9bc2d4e11ff289f0283/ops/c/src/core/ops_lazy.cpp#L935

Done it here
ops_exit_lazy(), misses freeing range and other parameter, added now

**Review by reguly — APPROVED** [review]

**Pull request merged**